opensourceleg/actuators/dephy.py

```diff
@@ -0,0 +1,1739 @@
+
+        if self.is_offline:
+            self.port = port
+        else:
+            Device.__init__(
+                self,
+                firmwareVersion=firmware_version,
+                port=port,
+                baudRate=baud_rate,
+                stopMotorOnDisconnect=stop_motor_on_disconnect,
+            )
+
+        self._debug_level: int = debug_level
+        self._dephy_log: bool = dephy_log
+
+        self._thermal_model: ThermalModel = ThermalModel(
+            temp_limit_windings=self.max_winding_temperature,
+            soft_border_C_windings=10,
+            temp_limit_case=self.max_case_temperature,
+            soft_border_C_case=10,
+        )
+        self._thermal_scale: float = 1.0
+
+        self._mode = CONTROL_MODES.VOLTAGE
+
+    def __repr__(self) -> str:
+        return f"{self.tag}[DephyActuator]"
+
+    @property
+    def _CONTROL_MODE_CONFIGS(self) -> CONTROL_MODE_CONFIGS:
+        return DEPHY_CONTROL_MODE_CONFIGS
+
+    @check_actuator_connection
+    def start(self) -> None:
+        try:
+            self.open()
+        except OSError as e:
+            print("\n")
+            LOGGER.error(
+                msg=f"[{self.__repr__()}] Need admin previleges to open the port '{self.port}'. \n\nPlease run the script with 'sudo' command or add the user to the dialout group.\n"
+            )
+            os._exit(status=1)
+
+        self.start_streaming(self._frequency)
+        self._data = self.read()
+
+        time.sleep(0.1)
+        # self._get_control_mode_config(self._mode).entry_callback(self)
+
+        default_mode_config = self._get_control_mode_config(self._mode)
+        if default_mode_config:
+            default_mode_config.entry_callback(self)
+
+    @check_actuator_stream
+    @check_actuator_open
+    def stop(self) -> None:
+        self.set_control_mode(mode=CONTROL_MODES.VOLTAGE)
+        self.stop_motor()
+
+        time.sleep(0.1)
+        self.close()
+
+    def update(self) -> None:
+
+        self._data = self.read()
+
+        self._thermal_model.T_c = self.case_temperature
+        self._thermal_scale = self._thermal_model.update_and_get_scale(
+            dt=1 / self.frequency,
+            motor_current=self.motor_current,
+        )
+        if self.case_temperature >= self.max_case_temperature:
+            LOGGER.error(
+                msg=f"[{str.upper(self.tag)}] Case thermal limit {self.max_case_temperature} reached. Stopping motor."
+            )
+            raise ThermalLimitException()
+
+        if self.winding_temperature >= self.max_winding_temperature:
+            LOGGER.error(
+                msg=f"[{str.upper(self.tag)}] Winding thermal limit {self.max_winding_temperature} reached. Stopping motor."
+            )
+            raise ThermalLimitException()
+        # Check for thermal fault, bit 2 of the execute status byte
+        if self._data["status_ex"] & 0b00000010 == 0b00000010:
+            raise RuntimeError("Actpack Thermal Limit Tripped")
```

Comment:
Should this change to your `Logger.error` style of error handling instead of my rough Runtime Error message?

Reply:
Added an error message and raised an exception with "Internal thermal limit tripped" message.